shub/image/build.py

```diff
@@ -71,6 +80,35 @@ def build_cmd(target, version, skip_tests):
         test_cmd(target, version)
 
 
+def _create_or_update_progress_bar(bar, event, verbose):
+    """Helper to handle build steps and track progress."""
+    step_row = BUILD_STEP_REGEX.match(event['stream'])
+    if step_row:
+        step_id, total = [int(val) for val in step_row.groups()]
+        if not bar:
+            bar = _create_progress_bar(total)
+        # ignore onbuild sub-steps
+        if step_id > bar.pos and bar.total == total:
+            bar.update()
+    return bar
+
+
+def _create_progress_bar(total):
+    return tqdm(
+        total=total,
+        desc='layers',
+        # XXX: click.get_text_stream or click.get_binary_stream don't
+        # work well with tqdm on Windows and Python 3
+        file=sys.stdout,
+        # helps to update bars on resizing terminal
+        dynamic_ncols=True,
+        # miniters improves progress on erratic updates caused by network
+        miniters=1,
+        # don't need rate here, let's simplify the bar
+        bar_format='{l_bar}{bar}| {n_fmt}/{total_fmt}',
+    )
```

Comment:
you can reuse https://github.com/scrapinghub/shub/blob/master/shub/image/push.py#L183 btw, say move to utils or something